File: StackQueue/JS/FunctionDev.java
package StackQueue.JS;

import java.util.ArrayList;
import java.util.List;

/**
 * FunctionDev
 * 8 점...
 */
public class FunctionDev {
    public int[] solution(int[] progresses, int[] speeds) {
        List<Integer> listProgresses = new ArrayList<>();
        for (int value : progresses) {
            listProgresses.add(value);
        }

        List<Integer> listSpeeds = new ArrayList<>();
        for (int value : speeds) {
            listSpeeds.add(value);
        }

        ArrayList<Integer> listDepNum = new ArrayList<>();

        while (listProgresses.size() > 0) {
            for (int i = 0; i < listProgresses.size(); i++) {
                listProgresses.set(i, listProgresses.get(i) + listSpeeds.get(i));
            }
            int count = 0;
            while (listProgresses.size() > 0 && listProgresses.get(0) >= 100) {
                listProgresses.remove(0);
                listSpeeds.remove(0);
                count++;
            }
            if (count > 0) {
                listDepNum.add(count);
            }
        }

        // ArrayList를 int[]로 변환
        int[] answer = new int[listDepNum.size()];
        for (int i = 0; i < answer.length; i++) {
            answer[i] = listDepNum.get(i);
        }
        return answer;
    }
    
}

File: StackQueue/GH/FunctionDev.py
# 기능개발 (7점)
def solution(progresses, speeds):
    
    answer = []
    
    # 현재 progresses 진행중인 인덱스
    curt_idx = 0
    
    last_idx = len(progresses) - 1
    
    while curt_idx <= last_idx :
        
        # 하루량 진행
        progresses = [ x + y for x, y in zip(progresses, speeds)]
        
        # 하루 완료량 계산
        count = 0
        
        while curt_idx <= last_idx and progresses[curt_idx] >= 100  :
            count += 1
            curt_idx += 1
            
        # 완료량이 0 이 아닌 경우 추가
        if count != 0:
            answer.append(count)
    
    

    return answer
